mongo_connector/doc_managers/doc_manager_base.py

```diff
@@ -55,6 +55,37 @@ def _retrieve_path(container, path, create=False):
                     raise ValueError
             return looking_at
 
+        def _set_field(doc, to_set, value):
+            if '.' in to_set:
+                path = to_set.split(".")
+                where = _retrieve_path(doc, path[:-1], create=True)
+                index = _convert_or_raise(where, path[-1])
+                wl = len(where)
+                if isinstance(where, list) and index >= wl:
+                    where.extend([None] * (index + 1 - wl))
+                where[index] = value
+            else:
+                doc[to_set] = value
+
+        def _unset_field(doc, to_unset):
+            try:
+                if '.' in to_unset:
+                    path = to_unset.split(".")
+                    where = _retrieve_path(doc, path[:-1])
+                    index_or_key = _convert_or_raise(where, path[-1])
+                    if isinstance(where, list):
+                        # Unset an array element sets it to null.
+                        where[index_or_key] = None
+                    else:
+                        # Unset field removes it entirely.
+                        del where[index_or_key]
+                else:
+                    del doc[to_unset]
+            except (KeyError, IndexError, ValueError):
+                # Ignore unset errors since MongoDB 2.4 records invalid
+                # $unsets in the oplog.
+                pass
```

Comment:
I'm a bit concerned that this could be masking legitimate problems. What if the destination has gone out of sync (e.g. rollback at destination) and we start applying updates with $unset to these documents? Before we would have logged an error, now we silently "apply" these changes, which could make recovery from this situation more difficult.

Perhaps we should consider if MongoDB 2.4 is worth supporting, since this behavior is incorrect, and this version has been end-of-lifed for awhile now. Alternatively, we could introduce yet another config option (*shudder*), or log this issue and continue (possibly unhelpful if we needed to recover from this situation).